### PR TITLE
Homework Cache Daniil Shadrin

### PR DESCRIPTION
**Pull request by rabbit72**

### CODE REVIEW

File: 05_concurrency/homework/cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	ticker := time.NewTicker(invalidationPeriod)
+
+	for {
+		select {
+		case <-ticker.C:
+			cache.invalidateExpired()
+		case _, ok := <-cache.quit:
+			if !ok {
+				return
+			}
```
Comment (andreipimenov) [inline 05_concurrency/homework/cache/cache.go:113]: This check looks weird. No one should send into `quit` but close it.

File: 05_concurrency/homework/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+func (cache *SimpleCache) Set(key string, value interface{}, ttl time.Duration) {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	tmpContainer, ok := cache.storage[key]
+
+	if !ok {
+		tmpContainer = valueContainter{}
+	}
```
Comment (andreipimenov) [inline 05_concurrency/homework/cache/cache.go:75]: cache.storage is a map of keys by structures. Thus reading will either return the value by key or empty value of its type which will be `valueContainer`. There is no need to check whether `ok` is true or false.

**Comment by andreipimenov** [comment]
LGTM

File: 05_concurrency/homework/cache/cache.go (hunk 1 above)
Comment (rabbit72) [reply 05_concurrency/homework/cache/cache.go:113]: Nobody sends. I use this channel only to close it when cache.Stop(). After closing the channel I stop runInvalidation goroutine.
Maybe there is a better way of how to send a "stop" signal.

**Pull request closed without merging**